src/vsr/replica.zig

```diff
@@ -658,9 +660,13 @@ pub fn ReplicaType(
             maybe(self.status == .recovering_head);
             if (self.status == .recovering) assert(self.solo());
 
-            // Asynchronously open the (Forest inside) StateMachine so that we can repair grid
-            // blocks if necessary:
-            self.state_machine.open(state_machine_open_callback);
+            // Asynchronously open the free set and then the (Forest inside) StateMachine so that we
+            // can repair grid blocks if necessary:
+            self.superblock.free_set_encoded.open(
+                &self.grid,
+                self.superblock.working.free_set_reference(),
+                free_set_open_callback,
+            );
```

Comment:
Heh, yeah I went back and forth here as well. I also have “context first, lambdas last” rule hard-wired in my brain, but it isn’t really applicable to zig, since there are no lambdas…

@jorangreef any tiger style insights about the order of the callback argument?

Reply:
I'm so glad you both raise this... because... I also reflexively put callbacks last in my mind. 😁 

"context first, lambdas last" makes sense also because we tend to "order according to logical use" and callbacks are the last thing that happens.

Beyond this, if you have a situation where you also have interim event callbacks that get emitted multiple times (e.g. as part of a streaming API) in addition to "done" callbacks, then putting the callback last follows the progression better.

So... let's do it, and start to move things over as we go.